### DISCUSSION TITLE
How to properly set secret values in values file

### DISCUSSION THREAD

**Discussion by some-crab** [apache/superset#27781] [Q&A / Help]
I would like to deploy superset on kubernetes with my docker-desktop environment. I am stuck on how to configure the values file, specifically with the passwords, which are sensitive data that i don't want to commit to github. I want to define all of my secret information in one place without committing it but I do not know how.

I have made a json file for my secrets and added a `secrets` field in the values file, so that I can run `helm upgrade --install --values helm/superset/values-custom.yaml superset superset/superset --set-json "secrets=$(cat helm/superset/secret_variables.json)"`. This has let me set the SECRET_KEY via `{{ .Values.secrets.secretKey }}`. But when it comes to the `postgresql.auth.password`, I am unable to template it. So then I decided to try this:

```
kubectl create secret generic my-db-secret  --from-literal=password=secretPassword
```

and then reference this in the values file with:
```yaml
postgresql:
  ...
  auth:
    existingSecret: my-db-secret
    ...
```
which it says in the comments that this will set the password. But instead the superset postgres pods fail to init, and when I check the logs of the pods I just see this:

```
Error from server (BadRequest): container "postgresql" in pod "superset-postgresql-0" is waiting to start: CreateContainerConfigError
```
Hardcoding the password is fine and the pods all start. The problem is with the secret. But I do not know enough about kubernetes to debug this. Seems like it might be a problem with the Bitnami PostgreSQL? 

But in any case, what I really want is a consistent way to store all of my superset secret values in a single place, and then inject them all into the values without writing a very long helm install command. What can I do to resolve this? Doesn't seem documented.

**Comment by MohamedRhimii** [comment]
@some-crab Hello, did you manage to solve this? I'm encountering the same exact issue with existingSecret parameter in postgres - superset.
Looking forward to your response!

**Reply by MohamedRhimii** [reply]
So I've made some modifications and made it work.
In order to do this, you need to:

1. Reference this section in values.yaml file:
`postgresql:
  ...
  auth:
    existingSecret: my-db-secret
    ...`
2. Create and deploy on kubernetes the "my-db-secret". It should be having these labels:
`  labels:
    app.kubernetes.io/instance: superset
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: postgresql
    helm.sh/chart: postgresql-x.x.x`
3. Be sure to NOT include the argocd label (if you have one), otherwise argocd will keep trying to prune your secret & keep the application status OutOfSync.

Hope this helps someone.
Best regards.

**Reply by AurimasNav** [reply]
You can have it in argocd as externalSecret and store it in some secretmanager.

**Reply by bodom0015** [reply]
This sounds very similar to what I'm trying to accomplish as well. We have an existing PostgreSQL instance (deployed via bitnami's Helm chart), and we would like to point Apache Superset at this existing database.

Our cluster already has the correct secret containing the database credentials, and the secret has been sealed (using SealedSecrets) and placed into source control:
```yaml
postgresql:
  ...    ...    ...    ...
  auth:
    username: REDACTED
    database: REDACTED
    existingSecret: "REDACTED-postgresql"
```

The `existingSecret` should contain the credentials that we need to access the DB, but there doesn't seem to be an elegant way to pass them into `.Values.supersetNode.connections.db_pass` (for example), which is used to build up [the secret-env](https://github.com/apache/superset/blob/75c500c9a53ce503b8636761f17b5b63eb8ee8e2/helm/superset/templates/secret-env.yaml#L47) later on 🤔 my assumption is that this `secret-env` file is needed, since it contains more than Postgres (some Redis secrets in there as well). Doing a quick search for `existingSecret` on this git repo, it looks like this value is never used outside of `values.yaml` which is likely not a good sign...

I will give the method you outlined above a try but am concerned that if the Helm chart is uninstalled, it may delete our existing database secret (which would be very bad, as other applications including the DB itself rely on its presence).
